### PR TITLE
Lazy gRPC connections

### PR DESCRIPTION
**Pull request by andrehp**

### CODE REVIEW

File: cluster/grpc_rpc_client.go [hunk 1]
```diff
@@ -106,7 +108,7 @@ func (gs *GRPCClient) Call(
 
 	parent, err := tracing.ExtractSpan(ctx)
 	if err != nil {
-		logger.Log.Warnf("failed to retrieve parent span: %s", err.Error())
+		logger.Log.Warnf("[grpc client] failed to retrieve parent span: %s", err.Error())
```
Comment (cscatolini) [inline cluster/grpc_rpc_client.go]: Shouldn't "grpcClient" be a field in the structured logging instead of a text?

Comment (andrehp) [reply cluster/grpc_rpc_client.go]: Yeah, ideally it should be, but pitaya's Logger interface doesn't expose the option to add fields :/

Comment (cscatolini) [reply cluster/grpc_rpc_client.go]: Oh noes,`[grpc client]` looks great then. Maybe something to improve in the future?

Comment (andrehp) [reply cluster/grpc_rpc_client.go]: Agreed

**Pull request merged**